File: school-managment/school-managment/src/main/java/com/alialperen/schoolmanagment/service/AdminService.java
package com.alialperen.schoolmanagment.service;

import java.util.List;

import com.alialperen.schoolmanagment.dto.FeeDto;
import com.alialperen.schoolmanagment.dto.SStudentDto;
import com.alialperen.schoolmanagment.dto.STeacherDto;
import com.alialperen.schoolmanagment.dto.StudentDto;
import com.alialperen.schoolmanagment.dto.StudentPermitDto;
import com.alialperen.schoolmanagment.dto.TeacherDto;

public interface AdminService {

	
	public void createAdminAccount();

	public StudentDto postStudent(StudentDto studentDto);

	public List<StudentDto> getAllStudents();
	
	public void deleteStudent(long studentId);
	
	public SStudentDto getStudentById(long id) throws Exception;
	
	public StudentDto updateStudent(Long studentId, StudentDto studentDto) throws Exception;

	public FeeDto payFee(long studentId, FeeDto feeDto);

	public List<StudentPermitDto> getAllAppliedPermits();

	public StudentPermitDto changePermitStatus(Long permitId, String status);

	public TeacherDto postTeacher(TeacherDto teacherDto);

	public List<TeacherDto> getAllTeachers();

	public void deleteTeacher(long teacherId);

	public STeacherDto getTeacherById(long teacherId)throws Exception;

	public TeacherDto updateTeacher(long teacherId, TeacherDto teacherDto) throws Exception;
}

File: school-managment-client/src/app/auth/services/admin/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { StorageService } from '../storage/storage.service';
import { Observable } from 'rxjs';



const BASIC_URL =  ["http://localhost:8080/api/admin/"]
@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  addStudent(studentDto:any):Observable<any>{
    return this.http.post<[]>(BASIC_URL+"student",studentDto,{
      headers: this.createAuthorizationHeader(),
    });

  }

  addActualite(actualite:any):Observable<any>{
    return this.http.post<[]>(BASIC_URL+"actualite",actualite,{
      headers: this.createAuthorizationHeader(),
    });

  }

  getAllStudents():Observable<any>{
    return this.http.get<[]>(BASIC_URL+"students",{
      headers:this.createAuthorizationHeader()
    })
  }

  deleteStudent(id:any):Observable<any>{
    return this.http.delete<[]>(BASIC_URL+`student/${id}`,{
      headers:this.createAuthorizationHeader()
    })
  }

  getStudentById(studentId:any):Observable<any>{
    return this.http.get<[]>(BASIC_URL+`student/${studentId}`,{
      headers:this.createAuthorizationHeader()
    })
  }

  updateStudent(studentId:number,studentDto:any):Observable<any>{
    return this.http.put<[]>(BASIC_URL+`student/${studentId}`,studentDto,{
      headers:this.createAuthorizationHeader(),
    });
  }

  getAllAppliedPermits():Observable<any>{
    return this.http.get<[]>(BASIC_URL+`permits`,{
      headers:this.createAuthorizationHeader()
    })
  }

  changePermitStatus(permitId:number,status:string):Observable<any>{
    return this.http.get<[]>(BASIC_URL+`permit/${permitId}/${status}`,{
      headers:this.createAuthorizationHeader()
    })
  }

  addTeacher(teacherDto:any):Observable<any>{
    return this.http.post<[]>(BASIC_URL+"teacher",teacherDto,{
      headers: this.createAuthorizationHeader(),
    });
  }

  deleteTeacher(teacherId:any):Observable<any>{
    return this.http.delete<[]>(BASIC_URL+`teacher/${teacherId}`,{
      headers:this.createAuthorizationHeader()
    })
  }

  getAllTeachers():Observable<any>{
    return this.http.get<[]>(BASIC_URL+"teachers",{
      headers:this.createAuthorizationHeader()
    })
  }

  getTeacherById(teacherId:any):Observable<any>{
    return this.http.get<[]>(BASIC_URL+`teacher/${teacherId}`,{
      headers:this.createAuthorizationHeader()
    })
  }

  updateTeacher(teacherId:number,teacherDto:any):Observable<any>{
    return this.http.put<[]>(BASIC_URL+`teacher/${teacherId}`,teacherDto,{
      headers:this.createAuthorizationHeader(),
    });
  }


  createAuthorizationHeader():HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      "Authorization","Bearer "+ StorageService.getToken()
    );
  }

  payFee(studentId:number,feeDto:any):Observable<any>{
    return this.http.post<[]>(BASIC_URL+`fee/${studentId}`,feeDto,{
      headers: this.createAuthorizationHeader(),
    });

  }
}
